lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -50,6 +52,12 @@ namespace {
   static cl::opt<bool> EnableBigQuery("souper-exhaustive-synthesis-enable-big-query",
     cl::desc("Enable big query in exhaustive synthesis (default=false)"),
     cl::init(false));
+  static cl::opt<bool> EnableDataflowPruning("souper-dataflow-pruning",
+    cl::desc("Enable pruning based on dataflow analysis (default=true)"),
+    cl::init(false));
+  static cl::opt<unsigned> DataflowPruningStatsLevel("souper-dataflow-pruning-stats-level",
```

Comment:
not sure this needs its own flag, why not piggyback on the existing souper debug flag?